src/lib.rs

```diff
@@ -311,56 +360,70 @@
 
 #![deny(non_camel_case_types)]
 
-extern crate url;
+#[macro_use]
+extern crate combine;
 extern crate sha1;
+extern crate url;
 
-#[cfg(feature="with-rustc-json")]
+extern crate net2;
+
+#[macro_use]
+extern crate futures;
+#[macro_use]
+extern crate tokio_io;
+extern crate tokio_reactor;
+extern crate tokio_tcp;
+
+#[cfg(feature = "with-rustc-json")]
 pub extern crate rustc_serialize as serialize;
-#[cfg(feature="with-unix-sockets")]
+#[cfg(feature = "with-async-unix-sockets")]
+extern crate tokio_uds;
+#[cfg(feature = "with-unix-sockets")]
 extern crate unix_socket;
 
 #[doc(hidden)]
-#[cfg(feature="with-rustc-json")]
+#[cfg(feature = "with-rustc-json")]
 pub use serialize::json::Json;
 
 // public api
-pub use parser::{parse_redis_value, Parser};
 pub use client::Client;
+pub use cmd::{cmd, pack_command, pipe, Cmd, Iter, Pipeline};
+pub use commands::{Commands, ControlFlow, PipelineCommands, PubSubCommands};
+pub use connection::{parse_redis_url, transaction, Connection, ConnectionAddr, ConnectionInfo,
+                     ConnectionLike, IntoConnectionInfo, Msg, PubSub};
+pub use parser::{parse_async, parse_redis_value, Parser};
 pub use script::{Script, ScriptInvocation};
-pub use connection::{Connection, ConnectionLike, ConnectionInfo, ConnectionAddr,
-                     IntoConnectionInfo, PubSub, Msg, transaction, parse_redis_url};
-pub use cmd::{cmd, Cmd, pipe, Pipeline, Iter, pack_command};
-pub use commands::{Commands, PipelineCommands, PubSubCommands, ControlFlow};
 
-pub use types::{
-    /* low level values */
-    Value,
+pub use types::{// utility functions
```

Comment:
rustfmt_skip appear to not work on this so I changed to // comments to sort of preserve the formatting